### PR TITLE
GroupResetSessionDAO - reset session applyContext on group url change.

### PR DESCRIPTION
**Pull request by jlhughes**
Presently only concerned with changes to group url. 
Can add more when required to the rule predicate.

### CODE REVIEW

File: src/foam/nanos/session/GroupResetSessionDAO.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                OR(
+                  EQ(Session.USER_ID, user.getId()),
+                  EQ(Session.AGENT_ID, user.getId())
+                ))
+              .select(new ArraySink())).getArray();
+            for ( Session session : sessions ) {
+              Session.APPLY_CONTEXT.clear(session);
+            }
+          }
+          group = group.findParent(x);
```
Comment (lchanmann) [inline src/foam/nanos/session/GroupResetSessionDAO.js]: Should it reset the session context of the descendent groups, when the url is not not set, as opposed to resetting the ancestors'?
Eg. url is set on sme group and business.group extends sme but doesn't have the url, when the url in sme group (the parent group) is updated then session context of the business user needs to be refreshed.

Comment (jlhughes) [reply src/foam/nanos/session/GroupResetSessionDAO.js]: No, this foam feature will not have knowledge of a business feature.

Comment (jlhughes) [reply src/foam/nanos/session/GroupResetSessionDAO.js]: but it could traverse in both directions, in general.

Comment (jlhughes) [reply src/foam/nanos/session/GroupResetSessionDAO.js]: but in that case, just easier to reset all sessions.

Comment (jlhughes) [reply src/foam/nanos/session/GroupResetSessionDAO.js]: and since that may be slow, probably better as an after rule so in it's own thread.

Comment (jlhughes) [reply src/foam/nanos/session/GroupResetSessionDAO.js]: refactored to rule, resets all sessions.

**Review by lchanmann — APPROVED** [review]

**Pull request merged**